### DISCUSSION TITLE
How can I inference with fp16

### DISCUSSION THREAD

**Discussion by zelabean** [Lightning-AI/pytorch-lightning#12781] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hello. I would like to infer a model trained with a trainer (precision=16) with fp16.
What should I do?

**Comment by rohitgr7** [comment]
```py
trainer = Trainer(precision=16)
trainer.test(...)
```
this doesn't work?

**Reply by zelabean** [reply]
I want use it on deployment environment. so just i want just using forward function like

with torch.no_grad():
     model.forward(input)
     
don't wanna use Trainer().

thank you for answer :)
